### PR TITLE
Protect routes with functional tests

### LINKED ISSUE #834: Protect routes with functional tests
Hi @jywarren @Tlazypanda @TildaDares 👋🏽 
For the functional tests, I noticed that we have about 13 controllers in the SW but we have functional tests for just about 9 controllers. So the following controllers are not being tested, device_controller, macros_controller, match_controller, and application_controller.

Just to be sure that I would be writing tests for these controllers?
Also, I was wondering what Macros in the macros_controller are?

Thank you! 🙏🏽

### PR DESCRIPTION
**Pull request by Georjane**
Fixes #834

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- `rake test`
* [ ] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [ ] pull request are descriptively named
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Hi @Georjane is this ready to merge do you think? @Tlazypanda does it conflict with any of the FTOs? If it's all good, we can regenerate the package.lock and yarn.lock and then it should be OK to merge!

**Comment by Georjane** [comment]
> Hi @Georjane is this ready to merge do you think? @Tlazypanda does it conflict with any of the FTOs? If it's all good, we can regenerate the package.lock and yarn.lock and then it should be OK to merge!

I think so @jywarren 
Waiting for @Tlazypanda to review and give me a feedback

File: app/controllers/macros_controller.rb [hunk 1]
```diff
@@ -13,7 +14,8 @@ def author
 
   def show
     @user = User.find_by_login params[:author]
-    @macro = Macro.find_by_title params[:id], conditions: { user_id: @user.id }, order: 'id DESC'
+    @macro = Macro.find_by_title params[:id]
```
Comment (Tlazypanda) [inline app/controllers/macros_controller.rb:17]: Hey @Georjane just going through this code, seems like removing this condition will essentially change what we are trying to fetch since we are not checking for user id so instead of finding a particular user's first record with title as "xyz" here, we will fetch the first record with this title without any relation to the user. Can you verify this once? 

I understand that this line was causing errors while executing can we instead rewrite it as - 
`@macros = Macro.find_by title: params[:id], conditions: { user_id: @user.id }, order: 'id DESC'`

File: app/controllers/spectrums_controller.rb [hunk 2]
```diff
@@ -206,7 +206,6 @@ def edit
   # replacing this with capture/save soon
   def create # rubocop:disable Metrics/AbcSize
     if logged_in? || params[:token] && User.find_by_token(params[:token])
-
```
Comment (Tlazypanda) [inline app/controllers/spectrums_controller.rb:209]: I think this file can be removed from the commit

File: app/views/spectrums/_list.html.erb [hunk 3]
```diff
@@ -9,8 +9,8 @@
         </a>
       <% end %>
       <a class="image visible-phone" style="float:left;width:50%;margin-right:10px;" href="<%= spectrum_path(spectrum.id) %>"><img src="<%= spectrum.photo.url(:thumb) %>" class="image" style="background:#ccc;"></a>
-      <h4><a data-transition="slide" href="<%= spectrum_path(spectrum.id) %>"><%=h truncate(spectrum.title,:length => 18) %></a></h4>
-      <p><a data-transition="slide" href="<%=h profile_path(spectrum.author) %>"><%=h spectrum.author %></a> | <%= time_ago_in_words(spectrum.created_at) %> ago<% if spectrum.like_count.positive? %> <i class="fa fa-star"></i> <%= spectrum.like_count %><% end %></p>
+      <h4><a data-transition="slide" href="<%= spectrum_path(spectrum.id) %>"><%= truncate(spectrum.title,:length => 18) %></a></h4>
+      <p><a class="author-profile" data-transition="slide" href="<%= profile_path(spectrum.author) %>"><%= spectrum.author %></a> | <%= time_ago_in_words(spectrum.created_at) %> ago<% if spectrum.like_count.positive? %> <i class="fa fa-star"></i> <%= spectrum.like_count %><% end %></p>
```
Comment (Tlazypanda) [inline app/views/spectrums/_list.html.erb:13]: I think this change popped up from another PR, is this relevant here? If not we can remove this :sweat_smile:

File: test/system/home_test.rb [hunk 4]
```diff
@@ -25,4 +25,15 @@ class HomeTest < ApplicationSystemTestCase
       assert_selector('a.btn.btn-large.btn-primary', text: 'Begin capturing »')
     end
   end
+
+  # test 'Check author profile link to be correct' do
```
Comment (Tlazypanda) [inline test/system/home_test.rb:29]: This change also seems to be from another PR :sweat_smile:

**Comment by Georjane** [comment]
@Tlazypanda I did all the required changes.
Thank you

**Comment by jywarren** [comment]
Great work @Georjane!!! Thanks for your persistence sticking with this one!!!
